amxmodx/amxmodx.cpp

```diff
@@ -252,16 +262,26 @@ static cell AMX_NATIVE_CALL client_print(AMX *amx, cell *params) /* 3 param */
 				g_langMngr.SetDefLang(i);
 				msg = format_amxstring(amx, params, 3, len);
 
+				// Client console truncates after byte 127.
+				// If format string is used, limit includes double new lines (125 + \n\n), otherwise one new line (126 + \n).
+				const auto bytesLimit = canUseFormatString ? 125 : 126;
+				
 				// params[2]: print_notify = 1, print_console = 2, print_chat = 3, print_center = 4
-				if (((params[2] == 1) || (params[2] == 2)) && (len > 126))	// Client console truncates after byte 127. (126 + \n = 127)
+				if (((params[2] == HUD_PRINTNOTIFY) || (params[2] == HUD_PRINTCONSOLE)) && (len > bytesLimit))	
 				{
-					len = 126;
+					len = bytesLimit;
 					if ((msg[len - 1] & 1 << 7))
 					{
 						len -= UTIL_CheckValidChar(msg + len - 1); // Don't truncate a multi-byte character
 					}
 				}
-				msg[len++] = '\n';	// Client expects newline from the server
+				msg[len++] = '\n';
+				
+				if (canUseFormatString && (!g_bmod_cstrike || (params[2] == HUD_PRINTNOTIFY || params[2] == HUD_PRINTCONSOLE)))
```

Comment:
This condition looks too complicated, can we split it up to make it more readable?